fix(ui): fixed upper jaw inline code string spacing

alt-shreya opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #52498

<!-- Feel free to add any additional description of changes below this line -->

In the [challenge CSS file](client/src/templates/Challenges/classic/editor.css), I added the `white-space` property to preserve whitespace and honor line breaks within the `<code>` element

![Updated_Spacing](https://github.com/freeCodeCamp/freeCodeCamp/assets/70858293/555144fd-9765-42e3-a882-f3f76763bf07)

alt-shreya committed:
fix(ui): updated CSS to add inline space

alt-shreya committed:
Merge branch 'main' into fix-ui-spacing

Sembauke commented:
<details><summary>Before:</summary>
<img width="587" alt="image" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/46919888/bbd78ebb-add0-4507-9a69-6e724ae71430">
</details> 

<details><summary>Now:</summary>
<img width="587" alt="image" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/46919888/dbac2597-e23e-4726-838b-2c19a1781c55">
</details>

alt-shreya committed:
Merge branch 'main' into fix-ui-spacing

alt-shreya commented:
@Sembauke Thanks for the feedback, will look into it :+1:

alt-shreya committed:
fix(ui): fixed extra spacing between lines

Sembauke commented:
I think we can remove the margin instead of setting the `line-spacing`:

```css
.editor-upper-jaw p, .editor-lower-jaw p{
  margin: 0;
}
```

alt-shreya committed:
fix(ui): changed size of margins

changed sizes of margins in `.editor-upper-jaw p, .editor-lower-jaw p` and `.description-container h1` classes

alt-shreya commented:
> I think we can remove the margin instead of setting the `line-spacing`:
> 
> ```css
> .editor-upper-jaw p, .editor-lower-jaw p{
>   margin: 0;
> }
> ```
> 
> Also a negative bottom margin on the description container to get the header and the description closer to getter would look better:
> 
> ```css
> .description-container h1 {
>   color: var(--secondary-color);
>   font-size: 1.1rem;
>   font-weight: 700;
>   line-height: 1.1;
>   margin: 0.1em 0 -0.6rem;
> }
> ```

made the changes, does this version look okay?

Sembauke commented:
Hey @alt-shreya,

It seems like the tests are failing, could you take a look?

alt-shreya committed:
Merge branch 'main' into fix-ui-spacing

alt-shreya commented:
> Hey @alt-shreya,
> 
> It seems like the tests are failing, could you take a look?

There seems to be an error with the landing page. I'm not sure why, or how to fix it.
@Sembauke could you help me resolve this?

```bash 
Landing page
    - Should render
    ✓ Has a visible large image on large viewports
```

alt-shreya committed:
Merge branch 'freeCodeCamp:main' into fix-ui-spacing

alt-shreya committed:
fix(css): fixed rendering

alt-shreya committed:
fix(css): updated margin size 

changed value of margin size to `rem` instead of `em`

alt-shreya committed:
Merge branch 'main' into fix-ui-spacing
